### PR TITLE
164892371-feature(signin): user should be able to sign in

### PR DESCRIPTION
**Pull request by nwamugo**
#### What does this PR do?
This PR allows a user to log into the Author's Haven app

#### Description of Task to be completed?
The feature collects user's email and password, checks to see if the user exists, and logs the user in

#### How should this be manually tested?
Clone the repo,
Install all dependencies,
Run npm run start:dev,
On Postman, post json/application inputs of email and password to localhost:3000/api/v1/users/login

#### Any background context you want to provide?
The user logging in allows the user to make the most of the application, as regards having a profile etc

#### What are the relevant pivotal tracker stories?(if applicable)
#164892371 User should be able to sign in

#### Screenshots
Screen Shot 2019-04-03 at 2 43 42 PM

### CODE REVIEW

File: src/routes/controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import dotenv from 'dotenv';
+
+import db from '../../db/models/index';
+import Secure from '../../utils/helper';
+
+dotenv.config();
```
Comment (Japhethca) [inline src/routes/controllers/auth.js]: please remove this, also line 1.

File: src/routes/controllers/auth.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import dotenv from 'dotenv';
+
+import db from '../../db/models/index';
+import Secure from '../../utils/helper';
```
Comment (Japhethca) [inline src/routes/controllers/auth.js]: let user `import { generate } from '../../utils/helper'`

File: src/routes/controllers/auth.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  async login(req, res) {
+    const { email, password } = req.body;
+    try {
+      const user = await User.findOne({
+        where: {
+          email,
+        }
+      });
+      if (!user) {
+        return res.status(404).json({
```
Comment (Japhethca) [inline src/routes/controllers/auth.js]: lets user our errorResponse helper function here.

File: src/utils/helper.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+
+const Secure = {
+  comparePassword(hashPassword, password) {
+    return bcrypt.compareSync(password, hashPassword);
+  },
+
+  generateToken(id) {
```
Comment (Japhethca) [inline src/utils/helper.js]: lets refactor this function to 
```javascript 
generateToken({ id: 1}, "3min")

function generateToken(data, time="30days") {
  const token = jwt.sign(
      data,
      process.env.SECRET_KEY, { expiresIn: time }
    );
   return token;
}
```

File: src/routes/controllers/auth.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import db from '../../db/models';
+import {
+  comparePassword,
+  generateToken,
+  successResponse,
+  errorResponse
+} from '../../utils/helpers';
+
```
Comment (Japhethca) [inline src/routes/controllers/auth.js]: nice on creating this file.

File: src/routes/auth.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ *       400:
+ *         description: Bad request.
+ *       401:
+ *         description: Authorization information is missing or invalid.
+ *       404:
+ *        description: A user with the specified ID was not found.
+ *       5XX:
+ *        description: Unexpected error.
+ */
+authRouter.post('/', createUser);
```
Comment (Japhethca) [inline src/routes/auth.js]: please can rename the name of this controller to `signup` and route to `/signup`

Comment (nwamugo) [reply src/routes/auth.js]: Please know that both signup and login are in the same controller.

For clarity, are you suggesting that auth.js be changed to signup.js @Japhethca ?

File: src/routes/controllers/auth.js (hunk 5 above)
Comment (nwamugo) [reply src/routes/controllers/auth.js]: Thank you.

File: src/routes/auth.js (hunk 6 above)
Comment (Japhethca) [reply src/routes/auth.js]: No, I meant the route and controller name.

Comment (nwamugo) [reply src/routes/auth.js]: Okay.

**Review by Japhethca — APPROVED** [review]

File: src/tests/routes/controllers/password.test.js [hunk 7]
```diff
@@ -9,14 +8,17 @@ const { expect } = chai;
 
 let resetToken;
 const userEmail = {
-  email: 'jsmith@gmail.com',
+  email: 'giant@gmail.com',
 };
 
+const fpUrl = '/api/v1/auth/users/forgot-password';
```
Comment (igbominadeveloper) [inline src/tests/routes/controllers/password.test.js:14]: Could you check this variable name so it depicts what you are trying to do?

**Pull request merged**